addons/stock/models/stock_move.py

```diff
@@ -500,15 +500,23 @@ def key_virtual_available(move, incoming=False):
             virtual_available_dict[key_context] = {res['id']: res['virtual_available'] for res in read_res}
 
         for move in product_moves:
+            if move.state == 'assigned':
+                move.forecast_availability = move.product_uom._compute_quantity(
+                    move.quantity, move.product_id.uom_id, rounding_method='HALF-UP')
+                continue
+            elif move.state == 'draft' and move.product_id.free_qty >= move.product_qty:
```

Comment:
```suggestion
            elif move.state == 'draft' and float_compare(move.product_id.free_qty, move.product_qty, float_rounding = move.product_id.uom_id.rounding)  >= 0:
```
Also, about the use of `product_id.free_qty`:
There already some prefetching done above to limit the computation of `_compute_quantities_dict()` which is pretty costly and context dependent for `virtual_available`.
Shouldn't something similar be done for the `free_qty` as well? :thinking: 
If anything, just to know that having free qty in `Warehouse B` won't imply having an available move in `Warehouse A`.

Reply:
you are right